tests/test_utils/gcp_system_decorator.py

```diff
@@ -0,0 +1,194 @@
+            if os.environ.get("AIRFLOW__CORE__UNIT_TEST_MODE"):
+                raise Exception("Please unset the AIRFLOW__CORE__UNIT_TEST_MODE")
+            variables = subprocess.check_output([ENV_FILE_RETRIEVER]).decode("utf-8")
+            print("Applying variables retrieved")
+            for line in variables.split("\n"):
+                try:
+                    variable, key = line.split("=")
+                except ValueError:
+                    continue
+                print("{}={}".format(variable, key))
+                os.environ[variable] = key
+
+
+RetrieveVariables.retrieve_variables()
+
+
+@contextmanager
+def gcp_authentication(service_key: str, project_extra: Optional[str] = None):
+    """
+    Context manager for invoking GCP credentials. Can be used in test for
+    managing resources with specified key.
+
+    **Example**: ::
+        def setUp():
+            with gcp_authentication('power_service_key.json'):
+                create_bucket()
+
+
+        @gcp_authentication('power_service_key.json')
+        def setUp():
+            create_bucket()
+
+    :param service_key: name of the service key that will be used to provide credentials
+    :type service_key: str
+    :param project_extra: optional project extras
+    :type project_extra: str
+    """
+    auth = GcpAuthenticator(gcp_key=service_key, project_extra=project_extra)
+    auth.gcp_authenticate()
+    yield auth
+    auth.gcp_revoke_authentication()
+
+
+def skip_gcp_system(
+    service_key: str,
+    long_lasting: bool = False,
+    require_local_executor: bool = False,
+    project_extra: Optional[str] = None,
+):
+    """
+    Decorator for skipping GCP system tests.
+
+    :param service_key: name of the service key that will be used to provide credentials
+    :type service_key: str
+    :param long_lasting: set True if a test take relatively long time
+    :type long_lasting: bool
+    :param require_local_executor: set True if test config must use local executor
+    :type require_local_executor: bool
+    :param project_extra: optional project extras
+    :type project_extra: str
+    """
+    if not os.environ.get("AIRFLOW_CONN_GOOGLE_CLOUD_DEFAULT"):
+        return unittest.skip(SKIP_NO_GCP_CONN_ID)
+
+    if GcpAuthenticator(service_key).full_key_path is None:
+        return unittest.skip(SKIP_TEST_WARNING)
+
+    if long_lasting and os.environ.get("GCP_ENABLE_LONG_TESTS") == "True":
+        return unittest.skip(SKIP_LONG_TEST_WARNING)
+
+    if require_local_executor and POSTGRES_LOCAL_EXECUTOR != os.environ.get(
+        "AIRFLOW_CONFIG"
+    ):
+        return unittest.skip(LOCAL_EXECUTOR_WARNING.format(POSTGRES_LOCAL_EXECUTOR))
+
+    def wrapper(cls):
+        # Add authenticator to test class
+        auth = GcpAuthenticator(gcp_key=service_key, project_extra=project_extra)
+        auth.gcp_store_authentication()
+        auth.gcp_authenticate()
+        auth.gcp_revoke_authentication()
```

Comment:
It is doing nothing? auth objects is not connected to cls.

Reply:
It just check if authentication works. Same thing was used in old test to assert everything will be ok. I think we can remove it, then in case of authentication any traceback will be more precise.